setup.py

```diff
@@ -98,7 +98,8 @@
         "planetary-computer~=1.0.0",
         "reretry~=0.11.8",
         "traceback-with-variables==2.0.4",
-        'scipy>=1.8' # used by sentinel-3 reader
+        'scipy>=1.8',  # used by sentinel-3 reader
+        "PyJWT[crypto]~=2.9.0",  # For identity tokens
```

Comment:
>= 2.10 dropped support for Python 3.8

Reply:
But we want to upgrade to py 3.11 eventually, were there is no reason I guess to stick to 2.9.0?
can't we leave that contraint open, or just specify `>=2.9.0`?

Reply:
I thought to pin it to avoid breaking it with updates but that is not a good reason to pin. My unittests should catch issues for breaking due to upgrades so I'll go for the `>=` suggestion

Reply:
indeed, I'm not a fan of putting too much constraints or hard pins in setup.py/pyproject.toml. In the long term you risk problems with unsolvable dependencies, impossible upgrades, or worse, security issues.

If there is clear need for pins that should go into a deploy `requirements.txt`